toolchain/sem_ir/function.h

```diff
@@ -15,38 +15,42 @@ namespace Carbon::SemIR {
 struct Function : public Printable<Function> {
   // A value that describes whether the function uses a return slot.
   enum class ReturnSlot : int8_t {
-    // Not yet known: the function has not been called or defined.
-    NotComputed,
     // The function is known to not use a return slot.
     Absent,
     // The function has a return slot, and a call to the function is expected to
     // have an additional final argument corresponding to the return slot.
     Present,
-    // Computing whether the function should have a return slot failed, for
-    // example because the return type was incomplete.
-    Error
+    // Computing whether the function should have a return slot failed because
+    // the return type was incomplete.
+    Incomplete,
+  };
+
+  // Information about how a function returns its return value.
+  struct ReturnInfo {
+    // Builds return information for a given declared return type.
+    static auto ForType(const File& file, TypeId type_id) -> ReturnInfo;
+
+    // Returns whether the return information could be fully computed.
+    auto is_valid() const -> bool { return slot != ReturnSlot::Incomplete; }
+
+    // Returns whether the function has a return slot. Can only be called for
+    // valid return info.
+    auto has_return_slot() const -> bool {
```

Comment:
FWIW, you might consider either `ReturnSlot slot` -> `ReturnSlot return_slot` or `has_return_slot` -> `has_slot`

Reply:
Done.